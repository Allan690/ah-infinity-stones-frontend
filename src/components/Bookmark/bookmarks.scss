.fav-btn {
    border: none;
    margin-right: 70px;
    cursor: pointer;
    i {
        color: #86C232 !important;
    }

    outline: none !important;
}

.fav-btn :hover {
    color: #86C232 !important;

}
