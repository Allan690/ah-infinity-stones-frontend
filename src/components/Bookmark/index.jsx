/* eslint-disable arrow-body-style */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Bookmark = ({ bookmark }) => {
  return (
    <Fragment>
      <div className='list-group'>
        <Link to={`/articles/${bookmark.article_slug}`} className='list-group-item list-group-item-action'>
          <div className='d-flex w-100 justify-content-between'>
            <h5 className='mb-1'>{ bookmark.article_title}</h5>
            <small>
              By &nbsp;
              <strong>
                {bookmark.article_author}
              </strong>
            </small>
          </div>
        </Link>
      </div>
    </Fragment>
  );
};

Bookmark.propTypes = {
  bookmark: PropTypes.shape({
    bookmark: PropTypes.object,
  }).isRequired,
};

export default Bookmark;
