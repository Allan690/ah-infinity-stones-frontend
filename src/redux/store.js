/* eslint no-underscore-dangle: 0 */
/* eslint-disable no-undef */
import { applyMiddleware, createStore, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import allReducers from './reducers';
import rootSaga from './sagas';

// only activate redux devtools on the browser otherwise just compose
const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const composeEnhancers = composer;

const middleware = createSagaMiddleware();
const store = createStore(
  allReducers,
  composeEnhancers(applyMiddleware(middleware)),
);

export default store;

middleware.run(rootSaga);
