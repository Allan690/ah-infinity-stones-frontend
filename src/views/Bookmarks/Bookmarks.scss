.loader {
  width: 80px;
  height:80px;
  top: 15%;
  border: 15px solid #f3f3f3;
  border-radius: 50%;
  border-bottom: 16px solid #86C232;
  border-top: 16px solid #222629;
  -webkit-animation: spin 2s linear infinite;
  animation: spin 2s linear infinite;
  margin: 0 auto;
}

#loader-container{
  margin: 0 auto;
  width: 20%;
}
.jumbotron {
  margin-top: 5%;
  width: 30%;
}
.card {
  width: 70%;
  margin: 0 auto;
}
