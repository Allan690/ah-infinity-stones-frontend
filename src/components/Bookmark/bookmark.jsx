import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bookmarkRequest } from '../../redux/actions/bookmarkActions';
import './bookmarks.scss';
import { successMessage, errorMessage } from '../../helpers/messages';

export class BookMark extends PureComponent {
  state = { className: 'fas fa-bookmark' };

  static propTypes = {
    bookmarkRequest: PropTypes.func.isRequired,
    slug: PropTypes.string.isRequired,
    message: PropTypes.string,
    error: PropTypes.string,
    prevMessage: PropTypes.string,
    prevError: PropTypes.string,
  };

  static defaultProps = {
    message: '',
    error: '',
    prevMessage: '',
    prevError: '',
  };

  componentDidUpdate(prevProps) {
    const { message, error } = this.props;
    const { prevMessage, prevError } = prevProps;
    if (message !== prevMessage && message !== null) {
      this.handleSuccess(message);
    } else if (prevError !== error) {
      this.handleError(error);
    }
  }

  handleBookMark = (event) => {
    const {
      slug, bookmarkRequest: bookmark,
    } = this.props;
    event.preventDefault();
    bookmark(slug);
  };

  handleSuccess = (message) => {
    successMessage(message);
    this.setState({ className: 'fas fa-bookmark' });
  };

  handleError = (error) => {
    errorMessage(error);
    this.setState({ className: 'far fa-bookmark' });
  };

  render() {
    const { className } = this.state;
    return (
      <div>
        <button
          type='button'
          className='fav-btn'
          onClick={this.handleBookMark}
        >
          <i className={className} />
          &nbsp;
          <strong>Bookmark</strong>
        </button>

      </div>
    );
  }
}

const mapDispatchToProps = { bookmarkRequest };

const mapStateToProps = (state) => {
  const { bookmarkReducer: { message, error } } = state;

  return { message, error };
};


export default connect(mapStateToProps, mapDispatchToProps)(BookMark);
