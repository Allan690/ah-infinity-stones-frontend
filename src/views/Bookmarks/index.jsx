/* eslint-disable arrow-body-style */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Bookmark from '../../components/Bookmark';
import { requestBookmarks } from '../../redux/actions/bookmarksActions';
import Loader from '../Loader';
import '../Login/Login.scss';
import './Bookmarks.scss';


class Bookmarks extends Component {
  componentDidMount() {
    const { requestBookmarks } = this.props;
    const isLoggedIn = localStorage.getItem('isLoggedIn');
    return isLoggedIn === 'true' && requestBookmarks();
  }

  render() {
    const { bookmarks } = this.props;
    const isLoggedIn = localStorage.getItem('isLoggedIn');
    const location = '/bookmarks';
    const bookmarksList = bookmarks && bookmarks.length ? (
      bookmarks.map((bookmark) => {
        return (
          <Bookmark key={bookmark.id} bookmark={bookmark} />
        );
      })
    ) : (<div id='loader-container' className='text-center'><Loader /></div>);
    return (
      isLoggedIn !== 'true' ? (
        <div className='container jumbotron text-center'>
          <Link to={{
            pathname: '/login',
            from: location,
          }}
          >
          Please log in
          </Link>
        </div>)
        : (
          <div className='card text-center'>
            <div className='card-header'>
              <h3>Bookmarked Articles</h3>
            </div>
            <div className='card-body'>
              {bookmarksList}
            </div>
          </div>
        )
    );
  }
}

Bookmarks.propTypes = {
  requestBookmarks: PropTypes.func.isRequired,
  bookmarks: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mapStateToProps = state => ({
  bookmarks: state.bookmarksReducer.bookmarks,
});

const actionCreators = { requestBookmarks };

export default connect(mapStateToProps, actionCreators)(Bookmarks);
