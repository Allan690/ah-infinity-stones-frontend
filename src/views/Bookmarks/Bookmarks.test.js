/* global it, expect, mount */
/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Bookmarks from '.';
import Bookmark from '../../components/Bookmark';
import store from '../../redux/store';

describe('Bookmarks Component', () => {
  let bookmarksContainer;
  let reduxBookmarks;
  let reduxBook;
  const bookmark = {
    article_title: 'test',
    article_author: 'justo1',
    article_slug: 'test',
  };
  beforeEach(() => {
    bookmarksContainer = shallow(<Bookmarks />);
    reduxBookmarks = mount(
      <Provider store={store}>
        <Router>
          <Bookmarks />
        </Router>
      </Provider>,
    );
  });

  it('Bookmarks renders successfully', () => {
    expect(reduxBookmarks.exists()).toEqual(true);
  });
  it('Should match snapshot', () => {
    expect(bookmarksContainer).toMatchSnapshot();
  });
  it('Bookmark mounts successfully', () => {
    reduxBook = mount(
      <Provider store={store}>
        <Router>
          <Bookmark bookmark={bookmark} />
        </Router>
      </Provider>,
    );
    expect(reduxBook.exists()).toEqual(true);
  });
});
