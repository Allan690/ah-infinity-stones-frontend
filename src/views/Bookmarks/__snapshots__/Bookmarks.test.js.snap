// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmarks Component Should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
